src/findFocalPlane.cpp

```diff
@@ -0,0 +1,184 @@
+// After using debug optics mode 4/simulate.py test 14, calculate
+// points of closest approach of reflected parallel photon beams,
+// to get approximate focal region. 
+#include <cstdlib>
+#include <iostream>
+
+// ROOT
+#include "TSystem.h"
+#include "TStyle.h"
+#include "TRegexp.h"
+#include "TCanvas.h"
+#include "TApplication.h"
+#include "TBox.h"
+#include "ROOT/RDataFrame.hxx"
+#include "TVector3.h"
+#include "TMatrix.h"
+#include "TMatrixD.h"
+#include "TMatrixDSparse.h"
+// edm4hep
+#include "edm4hep/MCParticleCollection.h"
+#include "edm4hep/SimTrackerHitCollection.h"
+
+using std::cout;
+using std::cerr;
+using std::endl;
+
+using namespace ROOT;
+using namespace ROOT::VecOps;
+using namespace edm4hep;
+
+TCanvas *CreateCanvas(TString name, Bool_t logx=0, Bool_t logy=0, Bool_t logz=0);
+std::vector<TVector3> d, e, fp, dirout;
+int nphotons = 0;
+int nAcc = 0;
+TVector3 focalPoint(std::vector<TVector3> dvec, std::vector<TVector3> endvec){
+  TMatrixD a(3,3), b(3,1);
+  TArrayI row(3),col(3);
+  
+  for(int i = 0; i < 3; i++) row[i] = col[i] = i;
+  TArrayD idarr(3); idarr.Reset(1.);
+  TMatrixDSparse identity(3,3);
+  identity.SetMatrixArray(3,row.GetArray(),col.GetArray(),idarr.GetArray());
+
+  for(int i = 0; i < dvec.size(); i++){
+    double darr[] = {dvec[i].X(),dvec[i].Y(),dvec[i].Z()};
+    double earr[] = {endvec[i].X(),endvec[i].Y(),endvec[i].Z()};
+
+    auto matd = TMatrixD(3,1,darr);  
+    auto mate = TMatrixD(3,1,earr);  
+    auto matdT = TMatrixD(matd);
+    matdT.T();
+    
+    a += (identity - matd*matdT);
+    b += (identity - matd*matdT)*mate;
+
+  }
+  auto c = (a.Invert()*b);
+  TVector3 x(TMatrixDRow(c,0)(0),TMatrixDRow(c,1)(0),TMatrixDRow(c,2)(0));
+  return x;
+};
+
+TVector3 avgDir(std::vector<TVector3> dvec){
+  double xavg = 0;
+  double yavg = 0;
+  double zavg = 0;
+  double n = 0;
+  for(int i = 0; i < dvec.size(); i++){
+    xavg += dvec[i].X();
+    yavg += dvec[i].Y();
+    zavg += dvec[i].Z();
+    n+=1.0;
+  }
+  
+  TVector3 outdir;
+  if(n > 0) outdir.SetXYZ(xavg/n, yavg/n, zavg/n);
+  return outdir;
+};
+
+int main(int argc, char** argv) {
+  // setup
+  TString infileN="sim_rich_vis.root";
```

Comment:
Use the default file (has been renamed since the last time you ran this)
```suggestion
  TString infileN="out/sim.edm4hep.root";
```